ledgerwallet/manifest_json.py

```diff
@@ -0,0 +1,60 @@
+import json
+import os
+from typing import Optional
+
+from ledgerwallet import params
+from ledgerwallet.manifest import AppManifest, icon_from_file
+from ledgerwallet.utils import get_device_name
+
+
+class AppManifestJson(AppManifest):
+    def __init__(self, filename):
+        with open(filename) as f:
+            self.path = os.path.dirname(filename)
+            self.dic = json.load(f)
+            assert "targetId" in self.dic and "binary" in self.dic
+
+    @property
+    def app_name(self) -> str:
+        return super().app_name
+
+    def data_size(self, device: str) -> int:
+        return int(self.dic.get("dataSize", 0))
+
+    def get_application_flags(self, device: str) -> int:
+        return int(self.dic.get("flags", "0"), 16)
+
+    def get_api_level(self, device: str) -> Optional[int]:
+        if "apiLevel" not in self.dic:
+            return None
+        return int(self.dic["apiLevel"], 10)
+
+    def get_binary(self, device: str) -> str:
+        return os.path.join(self.path, self.dic["binary"])
+
+    def serialize_parameters(self, device: str) -> bytes:
+        parameters = []
+        for entry, value in self.dic.items():
+            if entry == "name":
+                parameters.append({"type_": "BOLOS_TAG_APPNAME", "value": value})
+            elif entry == "version":
+                parameters.append({"type_": "BOLOS_TAG_APPVERSION", "value": value})
+            elif entry == "icon":
+                parameters.append(
+                    {"type_": "BOLOS_TAG_ICON", "value": icon_from_file(value)}
+                )
+            elif entry == "derivationPath":
+                derivation_paths = super().serialize_derivation_path(value)
```

Comment:
Just use `self.serialize_derivation_path(value)`: the inheritance freely gives you access to all methods of `AppManifest`, as by design `AppManifestJSON` **is** a `AppManifest`.
Use `super` whenever you override one of your motherclass method, but still want to exploit its behavior.

(Same comment applies to `AppManifestToml`)